### PR TITLE
Fix the overflow style for "Hide all checks"

### LINKED ISSUE #27928: After hiding and unhiding commit checks, there are only four checks available
### Description

take https://gitea.com/xorm/xorm/pulls/2364 for example. 
This PR has eight checks. After hiding and unhiding commit checks, there are only four checks on the web page.
Regression of https://github.com/go-gitea/gitea/pull/26284
cc @yp05327 

### Screenshots

Before: 
before
After:
after

### Gitea Version

current main

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

_No response_

### Browser Version

none

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #27928

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: web_src/css/repo.css [hunk 1]
```diff
@@ -3080,7 +3080,7 @@ tbody.commit-list {
 }
 
 .commit-status-list {
-  max-height: 195px; /* fit exactly 4 items */
+  max-height: 196px; /* fit exactly 4 items */
```
Comment (silverwind) [inline web_src/css/repo.css]: I guess the true value must lie somewhere in between 😆

File: web_src/js/features/repo-issue-pr-status.js [hunk 2]
```diff
@@ -2,11 +2,15 @@ export function initRepoPullRequestCommitStatus() {
   for (const btn of document.querySelectorAll('.commit-status-hide-checks')) {
     const panel = btn.closest('.commit-status-panel');
     const list = panel.querySelector('.commit-status-list');
+    let hasScrollBar = null;
     btn.addEventListener('click', () => {
+      if (hasScrollBar === null && !list.style.overflow) {
+        hasScrollBar = list.scrollHeight > list.clientHeight;
+      }
+      list.style.overflow = hasScrollBar ? '' : 'hidden'; // hide the flickering scrollbar when hiding if there was no scrollbar
```
Comment (silverwind) [inline web_src/js/features/repo-issue-pr-status.js]:
```suggestion
    let hasScrollBar = false;
    btn.addEventListener('click', () => {
      if (!hasScrollBar && !list.style.overflow) {
        hasScrollBar = list.scrollHeight > list.clientHeight;
      }
      list.style.overflow = hasScrollBar ? '' : 'hidden'; // hide the flickering scrollbar during transition if there was no scrollbar
```

Prefer boolean for vars starting with "has".

File: web_src/css/repo.css (hunk 1 above)
Comment (silverwind) [reply web_src/css/repo.css]: Also I think we should display 6 checks and reduce the vertical padding on each a bit. I can tweak this later.

Comment (silverwind) [reply web_src/css/repo.css]: Tweak done in https://github.com/go-gitea/gitea/pull/27932/commits/0a70fcc5d22483f8a9722b359568a6e1933c4c76:

image

File: web_src/js/features/repo-issue-pr-status.js (hunk 2 above)
Comment (silverwind) [reply web_src/js/features/repo-issue-pr-status.js]: Commited this in https://github.com/go-gitea/gitea/pull/27932/commits/ae0125d9660b29f4467fd8a7c73c14724cd2b6f5.

**Comment by silverwind** [comment]
Did some more fixes surrounding the borders. Now the header has the border, so it looks good even with MacOS elastic overscroll.

I have to say I find this JS less readable than before with the `data-toggled` property, but I can accept it.

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> I have to say I find this JS less readable than before with the `data-toggled` property, but I can accept it.

We can hugely simply it to something like  `data-toggled`, but the annoying scrollbar would appear.

> `  border: none !important;`

I think it is the right thing to do instead of `border-top/border-right/border-left`.

Each item had its own border-top, everything should be OK, there is no need to tweak `header.style.borderBottom`, and the `height` could be 39*4=234.

I do not understand the new change.

File: web_src/js/features/repo-issue-pr-status.js [hunk 3]
```diff
@@ -2,11 +2,21 @@ export function initRepoPullRequestCommitStatus() {
   for (const btn of document.querySelectorAll('.commit-status-hide-checks')) {
     const panel = btn.closest('.commit-status-panel');
     const list = panel.querySelector('.commit-status-list');
+    const header = panel.querySelector('.commit-status-header');
+    let hasScrollBar = false;
```
Comment (wxiaoguang) [inline web_src/js/features/repo-issue-pr-status.js]: This change is not right. The annoying scrollbar appears again during the transition when the list is short (click the show/hide quickly).

The old code is stable enough to handle the scrollbar flicker problem.

If it's difficult to understand and you can bear the UI flickering caused by the scrollbar, just remove this fine-tune.

Comment (silverwind) [reply web_src/js/features/repo-issue-pr-status.js]: I have overlay scrollbars on mac so it doesn't bother me that the scrollbar is visible during transition. Maybe show a video of what you mean?

Comment (wxiaoguang) [reply web_src/js/features/repo-issue-pr-status.js]: I guessed that you don't like flicker, so I added these tweaks. If it doesn't bother you, I will remove them.

**Comment by silverwind** [comment]
> I do not understand the new change.

On Mac, scroll down a bit then scroll up fast with elastic scrolling enabled. Previously it showed the border-top on the first item which was ugly. Now that the border is on the header and between items, it looks better.

File: web_src/js/features/repo-issue-pr-status.js (hunk 3 above)
Comment (silverwind) [reply web_src/js/features/repo-issue-pr-status.js]: Looks ok to me for the scrollbar to briefly appear:

toggle

**Comment by wxiaoguang** [comment]
> > I do not understand the new change.
> 
> On Mac, scroll down a bit then scroll up fast with elastic scrolling enabled. Previously it showed the border-top on the first item which was ugly. Now that the border is on the header and between items, it looks better.

Do you mean this? a03d597d4c3d6dcb5684168cfd1da44fe4fa1116

image

image

**Comment by silverwind** [comment]
Border-top issue has returned with latest change. Will fix. Sorry I can't even screenshot it, it's visible only when overscrolling with elastic scroll :)

**Review by silverwind — APPROVED** [review]

**Pull request merged**